### PR TITLE
Ability to continue training saved models

### PR DESCRIPTION
**Pull request by 6o6o**
Hey, thanks for the implementation. Great project, has a lot of potential.

I noticed train.py allows you to provide an input model with `-i` argument to train it further. However, it appears to be not implemented anywhere in the code except for argparse. I made an update to allow for model augmenting. Please review my changes and make your corrections if necessary.

Maybe there's a better way to determine starting epoch number, since reading it from a filename is not reliable. Perhaps there's some sort of embedded metadata in the model itself, that contains this info? Or passing it explicitly as a param would be better?

Thanks

### CODE REVIEW

**Comment by ttoinou** [comment]
Great I had the same idea but didn't know how to achieve this.
Did you test it ?

**Comment by 6o6o** [comment]
Yes, the loss stays on same levels as consecutive training. Not exactly the same, some variation occurs, but same in average.

Keep in mind, you'll have to know and provide exact same parameters as the input model was trained with to achieve same loss levels. Otherwise you'll end up blending different settings / images into the same model, unless this is what you want.

File: train.py [hunk 1]
```diff
@@ -85,7 +91,7 @@ def total_variation_regularization(x, beta=1):
 feature_s = vgg(Variable(style_b, volatile=True))
 gram_s = [gram_matrix(y) for y in feature_s]
 
-for epoch in range(n_epoch):
+for epoch in range(m_epoch, n_epoch):
```
Comment (sroecker) [inline train.py]: I would interpret n_epoch as n further epochs to train, not the total number.

I can confirm it works, although my trainings didn't get much better after 2 epochs.

Comment (6o6o) [inline train.py]: Seems reasonable, should I change the last argument to `m_epoch+n_epoch`?

**Comment by ttoinou** [comment]
> Otherwise you'll end up blending different settings / images into the same model, unless this is what you want.

Oh well, yes I want to do that, thanks the the information !
I don't have time to dig into the maths I don't understand everything yet..

File: train.py (hunk 1 above)
Comment (sroecker) [inline train.py]: Yeah I think that would make sense.

I also noticed that. A good approach is to write out checkpoints every few thousand images.

**Comment by yusuketomoto** [comment]
Thanks for PR.
I made some modifications. 
- save/load optimizer state
- exclude number of epochs from consideration(assuming a more general situation)

**Comment by 6o6o** [comment]
Thank you! This is exactly what I was looking for.
So the `-i` and `-r` should point to the same model file?

**Comment by yusuketomoto** [comment]
Yes. I suppose to be like `-i x.model -r x.state`.

**Pull request closed without merging**